control-plane/config/eventing-kafka-broker/200-controller/100-config-kafka-features.yaml

```diff
@@ -33,6 +33,10 @@ data:
     # 1. Enabled: KEDA autoscaling of consumers will be setup.
     # 2. Disabled: KEDA autoscaling of consumers will not be setup.
     controller.autoscaler: "disabled"
+    # The Go text/template used to generate consumergroup ID for triggers.
+    # The template can reference the trigger Kubernetes metadata only.
+    triggers.consumergroup.template: "knative-trigger-{{ .Namespace }}-{{ .Name }}"
   dispatcher.rate-limiter: "disabled"
   dispatcher.ordered-executor-metrics: "disabled"
   controller.autoscaler: "disabled"
+  triggers.consumergroup.template: "knative-trigger-{{ .Namespace }}-{{ .Name }}"
```

Comment:
Would it be possible to include the UID as suffix by default?

The reason being that it will allow users replaying events by deleting/recreating, since the trigger is almost fully mutable deleting/recreating wouldn't be necessary for anything other than replaying events while also providing a well known group id prefix for monitoring

Reply:
I see the comment on the issue:
> In addition, using the {{ . .Id }}, in iii you would still get a new consumergroup whenever the trigger is recreated, losing the offset again

it's not clear on why the need for recreating the trigger?

Reply:
I guess this is just a matter of preference. If somehow our CD would recreate the trigger, we want to keep the original consumergroup. Such a name would be very safe fo4 whenever k8s resources are destroyed.

> allow users replaying events by deleting/recreating
I think for "complex" use cases like these, we should let the user manually reset offsets, or perhaps create a new trigger with a different name for replay purposes.

But either is fine by me, since we can override the configuration for our preference.

Reply:
/lgtm
/approve

Thanks appreciate the additional thought